### ISSUE TITLE
Are you not going to make a n-pairs loss miner?

### ISSUE THREAD

**Issue by Heonjin** [KevinMusgrave/pytorch-metric-learning#109]
![image](https://user-images.githubusercontent.com/50353282/83103674-21557580-a0f2-11ea-8d1b-6c727541e3f0.png)
In the N-pairs loss paper, there is a "Hard negative class mining". But this is not implemented in this repository. Will you make this mining strategy? I just wonder it.

**Comment by nmll** [comment]
when use the N pair loss，it is increasing.may be which reason?

**Comment by KevinMusgrave** [comment]
@nmll Hard to say without more information. Perhaps your learning rate is too high?

**Comment by nmll** [comment]
Thanks，my learning rate is decaying.I do not use margin and miner,I use cos distance between the objects and then use the N pair loss,but the loss is increasing, I do not know why

**Comment by KevinMusgrave** [comment]
@nmll What are you passing into the N-pairs loss?

**Comment by nmll** [comment]
@KevinMusgrave  I pass into N-pairs loss the (anchors, labels,(anchor_index,pos_index,neg_index)),is the function required.

**Comment by KevinMusgrave** [comment]
@nmll Here are some other things you can try:

1. Try passing in just "anchors" and "labels":
```python
loss_func(anchors, labels)
```

2. See if the same thing happens with another loss function:
```python
from pytorch_metric_learning.losses import ContrastiveLoss
from pytorch_metric_learning.reducers import MeanReducer

loss_func = ContrastiveLoss(reducer=MeanReducer())
```

3. Start with a lower learning rate.

**Comment by nmll** [comment]
Thanks a lot!
